src/input_parsers/OnnxParser.cpp

```diff
@@ -411,12 +411,26 @@ Vector<int> getTensorIntValues( const onnx::TensorProto& tensor, const TensorSha
     {
         checkEndianness();
         const char* bytes = raw_data.c_str();
-        const int* ints = reinterpret_cast<const int*>( bytes );
-        for ( int i = 0; i < size; i++ )
-        {
-            int value = *(ints + i);
-            result.append( value );
-        }
+        onnx::TensorProto_DataType dataType = static_cast<onnx::TensorProto_DataType>( tensor.data_type() );
+        
+        if (dataType == onnx::TensorProto_DataType_INT32) {
+    		const int32_t* ints = reinterpret_cast<const int32_t*>( bytes );
+		for ( int i = 0; i < size; i++ )
+		{
+		    int value = *(ints + i);
+		    result.append( value );
+		}
+	} else if (dataType == onnx::TensorProto_DataType_INT64) {
+		const int64_t* ints = reinterpret_cast<const int64_t*>( bytes );
+		for ( int i = 0; i < size; i++ )
+		{
+		    int value = *(ints + i);
+		    result.append( value );
+		}
+	} else {
+		String errorMessage = Stringf( "Illegal data type for integer tensors used in the model. Only INT32 and INT64 supported." );
+            throw MarabouError( MarabouError::ONNX_PARSER_ERROR, errorMessage.ascii() );
+	}
```

Comment:
I think you also need to add a corresponding case for the `else` below when the data is not stored in `raw_data`. I'd advice lifting the `dataType` check above the `raw_data` check.